docs/developing.md

```diff
@@ -62,31 +62,158 @@ to use, or modify the examples to add your board.
 
 ## Adding a new driver
 
+Before writing a new sDDF device driver, it is highly recommended to first gain
+an understanding of the design principles behind the sDDF. All sDDF components,
+data structures and communication mechanisms are implemented to adhere to these
+principles, and consequently have a great deal in common. This is important
+whether the class of the new device driver exists or not, as non-adherence may
+cause difficulty when trying to utilise the new driver in other sDDF systems.
+
+The [sDDF design document](https://trustworthy.systems/projects/drivers/sddf-design.pdf)
+explains these principles and interfaces on a more abstract level and discusses
+why they were chosen.
+
+### Creations and modifications needed
+
 If you are adding a driver for an *existing* device class, you'll need to add
 the following:
-* the configuration file
+* the driver configuration file
 * the driver code itself
 * integration into the build system
 * integration into an example system
 
-### Writing the driver
+If your driver is not from an existing class, you'll additionally need to create
+the supporting infrastructure as well:
+* *virtualiser* protection domains to perform security operations as well as
+  client and data management tasks
+* an sDDF queue library containing the queue data structure and helper functions
+  used to pass data and buffers between drivers, virtualisers and clients
+* an sdfgen module allowing the new sub-system to be seamlessly added to systems
+
+Creating a new device class module for sdfgen is an involved task, and will
+require encoding into the tool all the information that is needed for the
+sub-system to be automatically generated. This includes:
+* necessary protection domains (driver, virtualisers)
+* memory regions and channels required for protection domains to use the
+  corresponding queue library
+* how virtualisers should organise their clients
+* the configuration options available for the system (memory region sizes,
+  protection domain priorities, client multiplexing options, data processing
+  options, etc)
```

Comment:
This section should just focus on adding a new driver to an existing device class since (especially for outside users) that will the majority of cases. We should still document adding a new device class but I think we should put that in a separate section and then link to it from here.

Reply:
It's probably worth moving the developing information to https://github.com/au-ts/ts_engineering_docs/tree/main - we've currently added such information for i2c / gpio / timer / spi and more are coming.

Reply:
I know, but there's a balance to be struck between the docs here and there. It's worth noting that we can make the other repo public because it's designed to avoid any confidential information. I'm not saying *don't put it here* however, it would just be nice for some of the details on serial as it is to be there!